### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by jhamman**
This is basically just a style/pep cleanup.  I removed a bunch of unused variables from internal modules.

I should say that I'm somewhat sheepishly submitting this PR, especially after seeing [this](https://youtu.be/wf-BqAjZb8M?list=PLP1xYYjBXksOm402NpkV1Ah5TRUKiQThn) talk by Raymond Hettinger.  However, since my [linter](https://atom.io/packages/linter-flake8) was constantly complaining about many of these, I figured I'd knock off all of them at once.  

There are quite a few line changes here but none of them should result in different behavior.

### CODE REVIEW

File: xray/core/pycompat.py [hunk 1]
```diff
@@ -17,13 +17,15 @@ def itervalues(d):
     from functools import reduce
     import builtins
     from urllib.request import urlretrieve
-else: # pragma: no cover
+else:  # pragma: no` cover
```
Comment (shoyer) [inline xray/core/pycompat.py]: extra ` character

File: xray/test/test_dataarray.py [hunk 2]
```diff
@@ -457,16 +456,6 @@ def test_isel_points(self):
         actual = da.isel_points(y=[1, 2], x=[1, 2], dim=['A', 'B'])
         assert 'points' in actual.coords
 
-    def test_isel_points(self):
```
Comment (shoyer) [inline xray/test/test_dataarray.py]: is this tested somewhere else?

File: xray/backends/netcdf3.py [hunk 3]
```diff
@@ -40,8 +40,8 @@ def coerce_nc3_dtype(arr):
         # TODO: raise a warning whenever casting the data-type instead?
         cast_arr = arr.astype(new_dtype)
         if ((('int' in dtype or 'U' in dtype) and
-                    not (cast_arr == arr).all())
-                or ('float' in dtype and
+                not (cast_arr == arr).all()) or
```
Comment (shoyer) [inline xray/backends/netcdf3.py]: Looks like your linter really doesn't like putting conjugations at the start of lines. Is that in PEP8?

File: xray/test/test_dataarray.py (hunk 2 above)
Comment (jhamman) [inline xray/test/test_dataarray.py]: Yes, we had a duplicate test in there (`test_isel_points` exists above and is much more thorough).

File: xray/backends/netcdf3.py (hunk 3 above)
Comment (jhamman) [inline xray/backends/netcdf3.py]: Yes, this is recommended in [PEP8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length):
>the preferred place to break around a binary operator is after the operator, not before it.

**Comment by shoyer** [comment]
OK, thanks!

**Comment by jhamman** [comment]
@shoyer - Thanks for merging.  How would you feel about adding [Lanscape](http://landscape.io) to our build checks?

If you haven't used Landscape before, its a style checker, much like what I was getting in my linter.  I ran it over my forks master branch to see what the output was like: https://landscape.io/github/jhamman/xray/2

**Comment by shoyer** [comment]
Sure, go ahead and set that up if you like. As long as it doesn't mark a build as failing, the info can't hurt. I'll note that I don't agree with many of it's "errors", though :). For example, as long as you write pure functions (which you should be doing anyways), it's perfectly fine to have mutable default argument.

**Pull request merged**